pkg/openapi/openapi_generated.go

```diff
@@ -1,4 +1,3 @@
-//go:build !ignore_autogenerated
```

Comment:
Please use go1.17 to `make generate`:

```
docker run --rm -v ${GOPATH}/src:/go/src -it golang:1.17.5 bash
# cd src/github.com/gardener/gardener
# make generate
```

Reply:
I have been using go v1.17 even locally. Basically, the issue was not with the version.

Ultimately, with the violation report I could figure out that a type with `[]string` cannot be used as it as. Instead, it needs to be indirectly referred to a list type which is why I introduced a new type called `type TaintKeys []string`. More details [here](https://github.com/kubernetes/kube-openapi/issues/175#issuecomment-546504117). This fixed violation error as well as the `make generate` errors. Thanks for the hint @ialidzhikov

Reply:
Ah, I see. `[]string` is actually wildly used in the types and I can see that for such fields we have reported violations. Then let's switch back to `[]string` again and live with the violation. 

PS. According to https://github.com/kubernetes/kube-openapi/issues/175#issuecomment-546504117 my understanding is that we rather have to specify list-type tags instead of doing such workaround with a new type. But let's leave this topic out of this PR.